src/shogun/mathematics/linalg/internal/implementation/Add.h

```diff
@@ -128,6 +129,7 @@ struct add<Backend::NATIVE, Matrix>
 	static void compute(T* A, T* B, T* C,
 		T alpha, T beta, index_t len)
 	{
+#pragma omp parallel for shared(A, B, C, alpha, beta)
```

Comment:
yeah I believe that makes sense. I tried locally with/without the omp directive for dot and the assembly it generated was same - compiler takes care of the vectorization.